### PR TITLE
Build vue files during deploy

### LINKED ISSUE #89: Set up config variables for API host url
@jhash I think I need to have you set this up, since I am too vague on how Vue reads the config vars. These are the existing config variables set through Heroku, accessible as `ENV['XXXX']` on the server. We can add others as needed, including `NODE_ENV`.

### Production

- RAILS_ENV=production

We need the host to be: `https://www.rapidskills.org`

### Staging

- RAILS_ENV=production
- RAPID_SKILLS_ENVIRONMENT=staging

We need the host to be: `https://rapid-skills-staging.herokuapp.com`

### Review apps

- RAILS_ENV=production
- RAPID_SKILLS_ENVIRONMENT=staging <-- we can change this to "review" if needed
- HEROKU_APP_NAME=xxx <- generated at the time the review app is created, will be random

We need the host to be: `https://[HEROKU_APP_NAME].herokuapp.com`

### Development

- RAILS_ENV=development

We need the host to be: `http://localhost:3000`. Looks like every developer can just set this locally in the `.env.local` file.

### PR DESCRIPTION
**Pull request by littleforest**
closes #89

### CODE REVIEW

File: Dockerfile.prod [hunk 1]
```diff
@@ -33,7 +33,16 @@ RUN bundle install --without development test
...
 
+# For ActiveAdmin assets
 # https://github.com/rails/rails/issues/32947#issuecomment-511532631
 RUN SECRET_KEY_BASE=dummy bin/rails assets:precompile
 
+# For puma server
 RUN mkdir -p tmp/pids
+
+# Build vue files
+COPY vue/package*.json vue/yarn.lock /usr/src/app/vue/
```
Comment (jhash) [inline Dockerfile.prod]: We shouldn't allow both package-lock.json and yarn.lock

Comment (jhash) [inline Dockerfile.prod]: Also, should you need to copy anything else if you already copied `.`?

File: config/environments/production.rb [hunk 2]
```diff
@@ -67,6 +67,8 @@
   host = ENV["HEROKU_APP_NAME"].present? ? "#{ENV['HEROKU_APP_NAME']}.herokuapp.com" : Rails.application.credentials.domain
   config.action_mailer.default_url_options = { host: host }
 
+  ENV["VUE_APP_API_BASE_URL"] = "https://#{host}"
```
Comment (jhash) [inline config/environments/production.rb]: Nice and easy. This happens before the dockerfile runs then?

File: vue/src/views/Home.vue [hunk 3]
```diff
@@ -3,7 +3,7 @@
     <div class="page--home__hero">
       <img alt="RapidSkills Logo" :src="APP_LOGO" class="page--home__hero__logo" />
       <div class="page--home__text page--home__hero__text page--home__hero__text--title">
-        Create new apprenticeship standards in a snap!
+        Create new apprenticeship standards in a snap FOO Dec 23 3:01pm!
```
Comment (jhash) [inline vue/src/views/Home.vue]: To be removed I assume

File: config/environments/production.rb (hunk 2 above)
Comment (littleforest) [reply config/environments/production.rb]: @jhash so when I added this commit https://github.com/WorkHands/RAPiDSkills/pull/131/commits/87c79ebda4940f9004c35d86cdab80f03d883cb5, I end up with URLS like: http://rapid-skills-favorites-wxsaqyz/undefined/standards. So it is clearly reading the base URL environment variable correctly, but I can't figure out where that undefined is coming from. When it was reading the base URL from the `vue/.env` file then there was no undefined.

File: Dockerfile.prod (hunk 1 above)
Comment (littleforest) [reply Dockerfile.prod]: @jhash which one are you using?

Comment (littleforest) [reply Dockerfile.prod]: Oh good point, I was trying to take advantage of the caching layer and copy the package/yarn files first, but I'm doing it after I copy everything so it doesn't make any sense. But I don't think anything is cached anyway with this Heroku setup and that it runs everything from scratch every time you deploy anyway, so it shouldn't matter.

File: vue/src/views/Home.vue (hunk 3 above)
Comment (littleforest) [reply vue/src/views/Home.vue]: Yes! Just testing that changes are getting built correctly :)

File: config/environments/production.rb (hunk 2 above)
Comment (jhash) [reply config/environments/production.rb]: @littleforest this is the only other thing in that URL but it's defined right before: ```export const API_NAMESPACE = '/api/v1';
const apiUrl = `${VUE_APP_API_BASE_URL}${API_NAMESPACE}`;```

**Comment by littleforest** [comment]
@jhash this is ready for re-review if you would like to take another look.

**Pull request merged**